tests/test_stack_viewer.py

```diff
@@ -0,0 +1,39 @@
+from pymmcore_plus import CMMCorePlus
+from useq import MDASequence
+
+from pymmcore_widgets._mda._datastore import QLocalDataStore
+from pymmcore_widgets._mda._stack_viewer import StackViewer
+
+sequence = MDASequence(
+    channels=[{"config": "DAPI", "exposure": 10}, {"config": "FITC", "exposure": 10}],
+    time_plan={"interval": 0.5, "loops": 3},
+    axis_order="tpcz",
+)
+
+
+def test_local(qtbot):
+    mmcore = CMMCorePlus.instance()
+    mmcore.loadSystemConfiguration()
+    datastore = QLocalDataStore(shape=(3, 1, 2, 512, 512), mmcore=mmcore)
+    canvas = StackViewer(mmcore=mmcore, datastore=datastore)
+    canvas.show()
+    qtbot.addWidget(canvas)
+    mmcore.run_mda(sequence)
+    if qtbot:
+        with qtbot.waitSignal(datastore.frame_ready, timeout=5000):
+            pass
+        with qtbot.waitSignal(datastore.frame_ready, timeout=5000):
+            pass
```

Comment:
i think just waiting for `events.sequenceFinished` would probably fix it here, but just to be safe, let's also use the non-threaded version (`mmcore.mda.run`) for the test:

```suggestion
    with qtbot.waitSignal(mmcore.mda.events.sequenceFinished):
        mmcore.mda.run(sequence)
```